ooniprobe/Model/Database/Result.h

```diff
@@ -2,7 +2,7 @@
 #import <SharkORM/SharkORM.h>
 #import "JsonResult.h"
 #import "Measurement.h"
-#define NOT_UPLOADED_QUERY @"is_failed = 0 AND (is_uploaded = 0 || report_id IS NULL)"
+#define NOT_UPLOADED_QUERY @"is_failed = 0 AND is_rerun = 0 AND is_done = 1 AND (is_uploaded = 0 || report_id IS NULL)"
```

Comment:
Did you add these checks because the underlying set could mutate as we process it?

Reply:
No, just because we don't care about re_run tests and of course the unfinished ones (they will never show up in results)

Reply:
Gotcha

Reply:
One more question: why don't we care about rerun tests?

Reply:
because a test who is re ran is usually a failed test and it has no json.
Furthermore is not visible in the list so it wil never get the upload bottom banner

Reply:
Right. Is it correct to say, though, that in principle a re-run test may not be failed? What happens in that case? Is the output file overwritten? Is this a corner case we need to care (maybe just by opening an issue and dealing with it in the future)?

Reply:
a test who has been re ran is failed.
Everything will be overwritten in the moment it will be re run, except the db row (a new one is created)

Reply:
Right, we can only re-run a test that failed. We may offer used to re-run a test in general in the future. When we will do that, we will probably need to think at the implications.